internal/broker/broker.go

```diff
@@ -126,13 +137,30 @@ type mcpBrokerImpl struct {
 	listeningMCPServer *server.MCPServer
 
 	logger *slog.Logger
+
+	//allowUnfilteredList
+	enforceToolFilter bool
 }
 
 // this ensures that mcpBrokerImpl implements the MCPBroker interface
 var _ MCPBroker = &mcpBrokerImpl{}
 
+type BrokerOpts struct {
+	EnforceToolFilter bool
+}
+
 // NewBroker creates a new MCPBroker
-func NewBroker(logger *slog.Logger) MCPBroker {
+func NewBroker(logger *slog.Logger, opts BrokerOpts) MCPBroker {
```

Comment:
Consider making BrokerOpts be an interface and the constructor `NewBroker(logger *slog.Logger, opts ...BrokerOpt)` so that we can add options and don't need to change so much code?  (Or keep it this way, just a suggestion).